src/components/survey/reusables.tsx

```diff
@@ -154,19 +154,18 @@ const Reusables: React.FC<Props> = ({ surveyId }) => {
    * 
    * @param updatedReusable updated reusable
    */
-  const onMaterialRowChange = async (newReusable: Reusable) => {
-    if (!keycloak?.token || !newReusable.id || !surveyId || newReusable.componentName === "") {
+  const onMaterialRowChange = async (updatedReusable: Reusable) => {
+    if (!keycloak?.token || !updatedReusable.id || !surveyId || updatedReusable.componentName === "") {
```

Comment:
Exclamation check should suffice in updatedReusable.componentName, since empty string is interpreted as "no value"
```suggestion
    if (!keycloak?.token || !updatedReusable.id || !surveyId || !updatedReusable.componentName) {
```